util/util.go

```diff
@@ -58,3 +63,50 @@ func AddImageNames(store storage.Store, image *storage.Image, addNames []string)
 	}
 	return nil
 }
+
+func ContextFromGlobalOptions(c *cli.Context, flagPrefix string) (*types.SystemContext, error) {
+	ctx := &types.SystemContext{
+		RegistriesDirPath: c.GlobalString("registries.d"),
+		DockerCertPath:    c.String(flagPrefix + "cert-dir"),
+		// DEPRECATED: keep this here for backward compatibility, but override
+		// them if per subcommand flags are provided (see below).
+		DockerInsecureSkipTLSVerify: !c.GlobalBoolT("tls-verify"),
+		OSTreeTmpDirPath:            c.String(flagPrefix + "ostree-tmp-dir"),
+	}
+	if c.IsSet(flagPrefix + "tls-verify") {
+		ctx.DockerInsecureSkipTLSVerify = !c.BoolT(flagPrefix + "tls-verify")
```

Comment:
Spent time looking at kpod and skopeo auth today.   Kpod does creds, cert-dir, signature-policy and tls-verify.  Skopeo does these (although signature-policy is just policy) plus ostree-temp-dir and registries.d.  I think for this PR I'm just going to emulate what kpod does and leave the other two off.  We can circle back later if necessary for the ostree-temp-dir and registries.d if needed.  

Fwiw, here's the secret decoder ring for the above.
creds:   un/pwd 
cert-dir path: Uses certificate at pathdiretory to connect to registry 
ostree-temp-dir:  Directory to use for OSTree temporary files 
registries.d: default directory containing registry information 
tls-verify: require https and verify certificates when talking to registry